bc_obps/reporting/service/report_attachment_service.py

```diff
@@ -43,3 +44,35 @@ def get_attachments(cls, report_version_id: int) -> QuerySet[ReportAttachment]:
     @classmethod
     def get_attachment(cls, report_version_id: int, attachment_id: int) -> ReportAttachment:
         return ReportAttachment.objects.get(report_version_id=report_version_id, id=attachment_id)
+
+    @classmethod
+    def save_attachment_confirmation(
+        cls,
+        report_version_id: int,
+        confirm_required_uploaded: bool,
+        confirm_existing_relevant: bool,
+    ) -> None:
+        """
+        Saves the attachment confirmation information for a supplementary report version.
+        """
+        default_confirmations = {
+            "confirm_supplementary_required_attachments_uploaded": confirm_required_uploaded,
+            "confirm_supplementary_existing_attachments_relevant": confirm_existing_relevant,
+        }
+
+        # Create or update the confirmation record
+        confirmation, created = ReportAttachmentConfirmation.objects.update_or_create(
+            report_version_id=report_version_id,
+            defaults=default_confirmations,
+        )
+
+    @classmethod
+    def get_attachment_confirmation(cls, report_version_id: int) -> ReportAttachmentConfirmation | None:
+        """
+        Retrieves the attachment confirmation record for a supplementary report version.
+        Returns None if no confirmation exists.
+        """
+        try:
+            return ReportAttachmentConfirmation.objects.get(report_version_id=report_version_id)
+        except ReportAttachmentConfirmation.DoesNotExist:
+            return None
```

Comment:
If we just let the system error out with the "DoesNotExist", django-ninja translates that to a 404 for us very nicely, no need to write additional error handling code